lib/ros/be/application/services/generator.rb

```diff
@@ -94,7 +95,28 @@ def kafka
 
           # kafka topics involved
           def kafka_topics
-            [fluentd.http_log_kafka_topic] + application.cloudevents_subjects
+            @kafka_topics ||= [fluentd.http_log_kafka_topic] + cloudevents_subjects
+          end
+
+          def cloudevents_subjects
+            subjects = []
+            platform_components.each do |service, definition|
+              ros_prefix = definition.config.ros ? "ros/" : ""
+              Dir.glob("#{ros_prefix}services/#{service.to_s}/doc/schemas/cloud_events/#{service.to_s}/*.avsc") do |file|
+                json = JSON.parse(File.read(file))
+                subjects.push(json["name"])
+              end
+            end
+            subjects
+          end
+
+          def platform_components
+           application.settings.components.platform.components.to_h.select{|k, v| v.nil? || v.dig(:config, :enabled).nil? || v.dig(:config, :enabled) }
+          end
```

Comment:
do we need to define this platform_components? since we already have `enabled_services` somewhere else.

Reply:
`enabled_services` gives us only the list of service names, without knowledge if service is Ros or not. So we do need actual platform services configs here...

However, there is the same code exists in the same file [below](https://github.com/rails-on-services/ros-cli/blob/57921df2518481fb8244d90ca0a9d357bdb917bb/lib/ros/be/application/services/generator.rb#L203).
But it is private to `Ros::Be::Application::Services::Generator`
Shall we make it public and declare a new instance of `Ros::Be::Application::Services::Generator` class just to access `platform_components`?

Something like this:
`generator=Ros::Be::Application::Services::Generator.new`
`generator.platform_components.each do |....|`